middleware/package.json

```diff
@@ -27,6 +27,7 @@
   "author": "TODO",
   "dependencies": {
     "@projectriff/message": "^1.0.0",
+    "@salesforce/core": "^2.10.0",
```

Comment:
I know @dhagberg-sf had some concerns about the SF core update. I'd like to know more about those. 

Other than the obvious space used for dealing with two versions (maybe 3 for the user function)...does the require system here work as we expect? I _think_ it does. This middleware will always use 2.10.0, the salesforce-sdk will use 2.1.6 and the user function will use whatever the latest that the user installs on init.

Obviously the versions may not play well together...but in this case the logger is specifically safe to use AFAICT. That might not be true in the future (like we pass a logger missing a function the older sdk expects or something). Also, the auto-complete for the user is likely at odds with the specific sdk objects we are providing to them compared to whatever is in their package.json.

Reply:
salesforce-sdk 1.3.1 now includes same version of core so /should/ be compatible - some testing will be needed.

Reply:
Pls update devDependencies' `typescript` to match the version used in `@salesforce/salesforce-sdk` also.

Reply:
:+1: I'll update to the latest version of the SDK and see if that's a viable path moving forward.